aip/general/0217.md

```diff
@@ -60,36 +71,45 @@ request with an error.
-there are unreachable parents or resources initially. Further, parents may
-alternate between being available and unavailable in unpredictable ways
-throughout the process of listing all the requested resources.
-
-These facts lead to the following guidance:
-
-- The response **must** provide any outstanding unreachable locations or
-  resources in the `unreachable` field on pages _following_ the final page that
-  contains a resource.
-  - The response **should not** include both requested data and unreachable
-    resources on the same page.
-    - For example, if there are two pages of books and one unavailable
-      publisher, there should be three pages total: first the two pages of
-      books, and then a final page with no books and the unavailable publisher.
-  - If the number of unreachable resources to list is very large, the response
-    **should** honor the `page_size` field in the same way as for resources. In
-    this case, all pages with requested information **should** precede all
-    pages with unavailable resources or locations.
-  - The final page's `unreachable` field **must** _only_ include resources or
-    parents that were partially provided (or missing completely) across the
-    entirety of the pagination process.
-    - For example, if a parent or resource was unreachable at the beginning of
-      pagination and it became reachable again and the entire set of previously
-      unreachable data was provided to the user on any page, the `unreachable`
-      field **must not** include the intermittently-unreachable parent or
-      resource.
-    - On the other hand, if only _some_ of the resources for a given parent are
-      provided during such an incident as described above, the parent or
-      resource **must** be included in the `unreachable` field.
+While preparing a page of results to fulfill a page fetch RPC e.g. an
+[AIP-132][aip-132] Standard List call, if the service encounters any unreachable
+resources or collections they **must** do the following:
+
+- Include the resource name for the unreachable resource in the `unreachable`
+  response field.
+  - The resource name **must** be the most appropriately scoped for the
+    unreachable resource or collection.
+    - For example, if a specific zone within a region is unreachable, the
+      unreachable resource name would be a zonal Location e.g.
+      `projects/example/locations/us-west1-a`, but if an entire region is
+      unreachable, the resource name would be a regional Location e.g.
+      `projects/example/locations/us-west1`.
+  - The resource name **must** be included, regardless of restrictive paging
+    paremeters e.g. `order_by`, when it is identified as unreachable.
+- Populate results that were previously considered unreachable on a following
+  page if their availability is restored and the paging parameters allow for
+  their inclusion.
+  - For example, if region `projects/example/locations/us-west1` was unavailable
```

Comment:
It may be worth documenting that the ordering could be implicit here - if the service is documented to return resources in resource name order when the client doesn't specify anything, that's sufficient to prevent the newly-available resources from being returned.

Reply:
I understand the scenario you are describing, but not sure what the suggestion is. I've included a bullet above this example to bring "documented default ordering" into scope of inclusion evaluation: https://github.com/aip-dev/google.aip.dev/pull/1392/commits/696b95d2d2d9b4aef02c28d11af99902dfc1ca1e

Is that what you mean?